graph/impl/src/main/java/org/jboss/windup/graph/GraphContextImpl.java

```diff
@@ -272,10 +274,19 @@ private TitanGraph initializeTitanGraph()
         // the performance decrease seems to be minimal.
         conf.setProperty("storage.berkeleydb.cache-percentage", 1);
 
+        conf.setProperty("storage.berkeleydb.lock-mode", LockMode.READ_UNCOMMITTED);
+        conf.setProperty("storage.berkeleydb.isolation-level", BerkeleyJEStoreManager.IsolationLevel.READ_UNCOMMITTED);
```

Comment:
That is interesting if it does improve memory usage. It is definitely a possibility.

As far as I know, all of our threading scenarios diligently avoid concurrent modification of graph data across threads due to the potential for deadlocks with Titan. I have not seen issues with this so far.